### DISCUSSION TITLE
CleanBlog: Specify a different navbar title from the page title

### DISCUSSION THREAD

**Discussion by VikeMK** [statiqdev/Discussions#103] [Q&A]
I have a statiq site using CleanBlog which has an "About Me" page that is linked in the navbar, I would like to make it so that the page's title in the navbar is **"About"** and the page's title when in the header is **"About Me"**. It seems that currently both of them are pulled from the same document property: `Title`, so there isn't a way to set them differently without overriding the template files. Is this something that could be added to the template? Or is there maybe a simpler way to achieve this?

**Comment by daveaglick** [comment]
> It seems that currently both of them are pulled from the same document property: `Title`

Yep, that's right on. More specifically, the theme file `_navbar.cshtml` that renders the navigation bar uses the HTML helper `@Html.DocumentLink()` to render the link to each top-level document. That helper does indeed use `Title` to get the text to use for the link.

> there isn't a way to set them differently without overriding the template files

Also true (at the moment at least). One nice thing is that because of the way the `input` folder takes precedence over the `theme` folder, you wouldn't have to actually edit the theme files to change this behavior. If you put a file with the same name in your `input` folder as in the `theme` folder, the one in your own `input` folder with override it. So one way to change this is to drop a new `_navbar.cshtml` file in your `input` folder with slightly different logic. This is actually exactly why the theme uses partials for some things like the navigation bar - to make them easier to customize.

In this case, you could introduce a new front matter property called something like `NavTitle` and then use that in your custom `_navbar.cshtml`:

```
<ul class="navbar-nav ml-auto">
    @foreach(IDocument doc in OutputPages.GetChildrenOf("index.html").Where(x => x.GetBool("ShowInNavbar", true)).OrderBy(x => x.GetInt(Keys.Order)))
    {
        // Don't add the home page to the nav bar
        if (doc.Destination != "index.html")
        {
            <li class="nav-item">
                @Html.DocumentLink(
                    doc,
                    doc.GetString("NavTitle", doc.GetTitle()),
                    new Dictionary<object, object> { { "class", "nav-link" } })
            </li>
        }
    }
</ul>
```

> Is this something that could be added to the template?

That's a good question. It would be trivial to change the official `_navbar.cshtml` file in the CleanBlog theme to match the above and it seems like a good little enhancement. I'll go ahead and update it now.

**Reply by daveaglick** [reply]
Update CleanBlog, ended up calling the new metadata `NavbarTitle`
